### DISCUSSION TITLE
Boundary conditions and external torque implementation for a simply supported rod

### DISCUSSION THREAD

**Discussion by Srishti-Siddharth-iitb** [GazzolaLab/PyElastica#433] [Q&A] [answered]
For one of my applications, I wish to simulate the Cosserat rod with the following boundary conditions:

1. Base of the rod: Position $r=[0;0;0]$ is fixed, the orientation $Q \in $ SO(3) is free, and there is an external torque (only at the base) to rotate the rod. 
2. Tip of the rod: free, no external forces or torques.

I have attached a figure explaining the setup. Could you please let me know how to simulate this rod using PyElastica?

![Rotating_rod](https://github.com/user-attachments/assets/ca658d2d-a3cd-4f7b-ab10-401e603c4d9a)

**Comment by armantekinalp** [comment, marked as answer]
Hey 

You can use [GeneralConstraints ](https://github.com/GazzolaLab/PyElastica/blob/master/elastica/boundary_conditions.py#L239) as a boundary condition in PyElastica. Please see the documentation.

For the rotating torque you have to write a custom forcing class for you application. Please see [EndPointForces](https://github.com/GazzolaLab/PyElastica/blob/master/elastica/external_forces.py#L110) class here. Although this forcing classs applies tip force, implementation wise it is very similar to your application.

**Comment by armantekinalp** [comment]
I am converting this to a discussion since it is not an issue.

**Reply by Srishti-Siddharth-iitb** [reply]
Thank you. I have another follow-up question: do we have to add an external force or an external couple to simulate the torque at the base?
